extensions/typescript-language-features/src/languageFeatures/diagnostics.ts

```diff
@@ -194,6 +200,37 @@ class DiagnosticsTelemetryManager extends Disposable {
 		this._registerTelemetryEventEmitter();
 	}
 
+	public logDiagnosticsPerformanceTelemetry(performanceData: DiagnosticPerformanceData[]): void {
+		performanceData.forEach(data => {
+			/* __GDPR__
+				"diagnostics.performance" : {
+					"owner": "mjbvz",
+					"syntaxDiagDuration" : { "classification": "SystemMetaData", "purpose": "FeatureInsight" },
+					"semanticDiagDuration" : { "classification": "SystemMetaData", "purpose": "FeatureInsight" },
+					"suggestionDiagDuration" : { "classification": "SystemMetaData", "purpose": "FeatureInsight" },
+					"regionSemanticDiagDuration" : { "classification": "SystemMetaData", "purpose": "FeatureInsight" },
+					"fileLineCount" : { "classification": "SystemMetaData", "purpose": "FeatureInsight" },
+					"${include}": [
+						"${TypeScriptCommonProperties}"
+					]
+				}
+			*/
+			this._telemetryReporter.logTelemetry('diagnostics.performance',
+				{
+					// @ts-expect-error until ts 5.6
+					syntaxDiagDuration: data.syntaxDiag,
```

Comment:
I think all of these numbers should be measures instead of normal telemetry properties. Just add `"isMeasurement": true` to each property definition